daemons/fenced/fenced_commands.c

```diff
@@ -3021,10 +3023,13 @@ remove_relay_op(xmlNode * request)
                     }
                 }
             }
-            crm_debug("Deleting relay op %s ('%s' targeting %s for %s), "
-                      "replaced by op %s ('%s' targeting %s for %s)",
-                      relay_op->id, relay_op->action, relay_op->target,
-                      relay_op->client_name, op_id, relay_op->action, target,
+            crm_debug("Deleting relay op %s ('%s'%s%s for %s), "
+                      "replaced by op %s ('%s'%s%s for %s)",
+                      relay_op->id, relay_op->action,
+                      (relay_op->target == NULL)? "" : " targeting ",
+                      pcmk__s(relay_op->target, ""),
+                      relay_op->client_name, op_id, relay_op->action,
+                      (target == NULL)? "" : " targeting ", pcmk__s(target, ""),
                       client_name);
```

Comment:
that is actually a recurring pattern we could cover with a macro like pcmk__s_if_s evaluating to 2 strings.

Reply:
We do have PCMK__OP_FMT which consumes 3 specifiers so maybe something like

    PCMK__SS_FMT(" targeting ", target)

defined like

    #define PCMK__SS_FMT(label, var) ((var == NULL)? "" : label), (var == NULL)? "" : var)

?

Reply:
yep ... that is about what I had in mind.
even if a double s appears to be the natural naming I tend to avoid those though.

Reply:
I'd like to stick with the `PCMK__*_FMT` naming to hint that it consumes more than one specifier ... do you have any suggestions for a short middle part?

Reply:
maybe SIFS ... of course there is some beauty in doing it like C++ name-mangling ...

Reply:
I got confused -- `PCMK__OP_FMT` is for the format string (the "%s%s" part here). I don't think we have any equivalent macros for format arguments yet.

What do you think of `PCMK__LABELED_S(" targeting ", target)` or `PCMK__S_ELSE_EMPTY(" targeting ", target)`? It's hard to come up with something short that's not confusing ...

Reply:
All have their pros and cons. I still like my S_IF_S being so short. What about LABELED_S_IF_S - bit longer but has the full logic.